File: 75-sort-colors/75-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        // sort(nums.begin(),nums.end());
        int n=nums.size();
        int lo=0,mid=0,hi=n-1;
        for(int i=0;i<n;i++)
        {
            if(nums[mid]==0)
            {
                swap(nums[mid],nums[lo]);
                mid++;
                lo++;
            }
             else if(nums[mid]==2)
            {
                swap(nums[mid],nums[hi]);
                hi--;
            }
            else if(nums[mid]==1)
            {
                mid++;
            }
        }
    }
};

File: 75-sort-colors/75-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n=int(len(nums))
        lo=0
        hi=n-1
        mid=0
        temp=0
        for i in range(0,n):
            if(nums[mid]==0):
                temp=nums[mid]
                nums[mid]=nums[lo]
                nums[lo]=temp
                lo+=1
                mid+=1
            elif(nums[mid]==2):
                temp=nums[mid]
                nums[mid]=nums[hi]
                nums[hi]=temp
                hi-=1
            else:
                mid+=1